### PR TITLE
Nirms Validation + Tests

### PR DESCRIPTION
**Pull request by Ewan-James-Donovan**

### CODE REVIEW

**Review by besscerule — DISMISSED** [review]
LGTM

File: app/forms/NirmsNumberFormProvider.scala [hunk 1]
```diff
@@ -26,6 +26,6 @@ class NirmsNumberFormProvider @Inject() extends Mappings {
   def apply(): Form[String] =
     Form(
       "value" -> text("nirmsNumber.error.required")
-        .verifying(maxLength(100, "nirmsNumber.error.length"))
+        .verifying(regexp("RMS-?(GB|NI)-?[0-9]{6}", "nirmsNumber.error.invalidFormat"))
```
Comment (ChrisCookOC) [inline app/forms/NirmsNumberFormProvider.scala:30]: Will this allow whitespace in the answers? Might need that whitespace helper if not

File: test/forms/behaviours/FieldBehaviours.scala [hunk 2]
```diff
@@ -55,4 +63,18 @@ trait FieldBehaviours extends FormSpec with ScalaCheckPropertyChecks with Genera
       result.errors mustEqual Seq(requiredError)
     }
   }
-}
+
+  def fieldThatErrorsOnInvalidData(
```
Comment (ChrisCookOC) [inline test/forms/behaviours/FieldBehaviours.scala:52]: This is being defined twice?

**Review by ChrisCookOC — CHANGES_REQUESTED** [review]

File: test/forms/behaviours/FieldBehaviours.scala (hunk 2 above)
Comment (Ewan-James-Donovan) [reply test/forms/behaviours/FieldBehaviours.scala:52]: Done!

File: app/forms/NirmsNumberFormProvider.scala (hunk 1 above)
Comment (Ewan-James-Donovan) [reply app/forms/NirmsNumberFormProvider.scala:30]: Added it :)

**Review by ChrisCookOC — APPROVED** [review]

**Pull request merged**